### PR TITLE
Add go to line shortcut in shortcuts modal

### PR DESCRIPTION
**Pull request by pradeepgangwar**
Associated Issue: #4804 

### Summary of Changes

* Add go to line shortcut `CtrlorCmd+:` in shortcut modal.

### Screenshots

screenshot-2017-12-16 debugger 2

### CODE REVIEW

File: assets/panel/debugger.properties [hunk 1]
```diff
@@ -173,6 +173,10 @@ projectTextSearch.key=CmdOrCtrl+Shift+F
 # modal for searching functions in a file.
 functionSearch.key=CmdOrCtrl+Shift+O
 
+# LOCALIZATION NOTE (gotoLine.key): A key shortcut to jump to
+# a specific line in a file.
+gotoLine.key=CmdOrCtrl+:
```
Comment (jasonLaster) [inline assets/panel/debugger.properties]: it looks like we have `gotoLineModal.key` is that the same thing?

Comment (pradeepgangwar) [reply assets/panel/debugger.properties]: @jasonLaster Looks like there are different keys for Mac and Windows/Linux platform. I use linux and `Ctrl+:` is the key for opening that modal. Should i add `Cmd+Shift+;` also. 😃

Comment (jasonLaster) [reply assets/panel/debugger.properties]: Hmm, lets look at the code because shift should not be platform dependent

**Comment by jasonLaster** [comment]
Hey @pradeepgangwar can you confirm that you're seeing `Cmd+Shift+;` on linux? It's really weird to me that that is the case.

Perhaps you can add a breakpoint in App.js and make sure that we don't have a busted event listener...

Also, @nyrosmith @GarethSharpe @Garbee what do you see on linux?

**Comment by pradeepgangwar** [comment]
In linux `CtrlOrCmd + ;` works for gotoLine Modal. I checked and found that `CtrlOrCmd+;` triggers function https://github.com/devtools-html/debugger.html/blob/master/src/components/App.js#L121 . I changed key combination and found new combination to be working fine. Maybe linux interprets this particular combination differently.

**Comment by jasonLaster** [comment]
i think ctrl+shift+: also worked for the mac. maybe we don't preclude that...

**Comment by jasonLaster** [comment]
Ping @pradeepgangwar

**Comment by pradeepgangwar** [comment]
I think we can have `Ctrl+Shift+;` for now in shortcut modal. It that is ok I will update my PR in a day .

**Comment by jasonLaster** [comment]
hmm, if we support both... then shouldn't we do `Ctrl+;`

**Comment by pradeepgangwar** [comment]
screenshot-2018-1-6 debugger

Please review the changes.

File: src/components/ShortcutsModal.js [hunk 2]
```diff
@@ -86,6 +86,14 @@ export class ShortcutsModal extends Component<Props> {
           L10N.getStr("shortcuts.functionSearch"),
           formatKeyShortcut(L10N.getStr("functionSearch.key"))
         )}
+        {this.renderShorcutItem(
+          L10N.getStr("shortcuts.gotoLine"),
+          formatKeyShortcut(L10N.getStr("gotoLineModal.key"))
+        )}
+        {this.renderShorcutItem(
```
Comment (jasonLaster) [inline src/components/ShortcutsModal.js]: I think we can check if we're on linux. there is definitely a windows or mac check elsewhere

**Comment by pradeepgangwar** [comment]
@jasonLaster Sorry, I completely forgot about the platform checks. I changed it to have platform checks. Also I capitalized first letter of Shortcut string in welcome box. I am not sure of what works on windows as go to Line shortcut key.

File: assets/panel/debugger.properties [hunk 3]
```diff
@@ -542,16 +542,16 @@ watchExpressions.refreshButton=Refresh
 # LOCALIZATION NOTE (welcome.search): The center pane welcome panel's
 # search prompt. e.g. cmd+p to search for files. On windows, it's ctrl, on
 # a mac we use the unicode character.
-welcome.search=%S to search for sources
+welcome.search=%S To search for sources
```
Comment (darkwing) [inline assets/panel/debugger.properties]: Let's keep this lowercase for sake of consistency.

File: assets/panel/debugger.properties [hunk 4]
```diff
@@ -542,16 +542,16 @@ watchExpressions.refreshButton=Refresh
...
 
 # LOCALIZATION NOTE (welcome.findInFiles): The center pane welcome panel's
 # search prompt. e.g. cmd+f to search for files. On windows, it's ctrl+shift+f, on
 # a mac we use the unicode character.
-welcome.findInFiles=%S to find in files
+welcome.findInFiles=%S To find in files
```
Comment (darkwing) [inline assets/panel/debugger.properties]: Let's keep this lowercase for sake of consistency.

File: assets/panel/debugger.properties [hunk 5]
```diff
@@ -542,16 +542,16 @@ watchExpressions.refreshButton=Refresh
...
 
 # LOCALIZATION NOTE (welcome.searchFunction): Label displayed in the welcome
 # panel. %S is replaced by the keyboard shortcut to search for functions.
-welcome.searchFunction=%S to search for functions in file
+welcome.searchFunction=%S To search for functions in file
```
Comment (darkwing) [inline assets/panel/debugger.properties]: Let's keep this lowercase for sake of consistency.

**Comment by darkwing** [comment]
This is a really great start!  I did `Command+Shift+;` on my Mac, the `:` was there, I entered a line number, hit `Enter`, and the code pane went to the line I desired and highlighted it!  Awesome!

A few issues I'm seeing:

1.  You can still use `Command+Shift+;` (the pane slides in) when there are no sources opened.  We shouldn't allow line search if there's no source open.

2.  The autocomplete listing shows file names.  I'm guessing we don't want any autocomplete if someone is looking for a line number.  Or if we want to autocomplete, maybe we show a preview of the line contents.  Either seem cool.

**Comment by jasonLaster** [comment]
Ping @pradeepgangwar do you have a chance to work on this in the near future?

**Comment by pradeepgangwar** [comment]
@jasonLaster please let me know changes needed for this PR. I fixed earlier reviews  regarding identification of linux or mac.

**Comment by Garbee** [comment]
`Ctrl +` and `Ctrl =` are often used in a lot of software for zooming in applications. Since zooming is an accessibility tool perhaps we shouldn't go overriding this expected behavior.

Why not align with Chrome's shortcuts here? Use `ctrl p` and when the user types `:` to start, switch to line jump mode? This keeps things consistent (making it easier for people switching toolsets) and doesn't override expected accessibility controls.

**Comment by darkwing** [comment]
Hey @pradeepgangwar !  Suggested changes are in this comment:

https://github.com/devtools-html/debugger.html/pull/4932#issuecomment-356806341

Thank you!  Let me know if you have questions!

**Comment by jasonLaster** [comment]
@pradeepgangwar I opted to simplify the logic a bit. This should be ready to go. Sorry for the delay

File: assets/panel/debugger.properties [hunk 6]
```diff
@@ -704,10 +704,10 @@ functionSearchSeparatorLabel=←
 
 # LOCALIZATION NOTE(gotoLineModal.placeholder): The placeholder
 # text displayed when the user searches for specific lines in a file
-# Do not localize "CmdOrCtrl+Shift+;", or change the format of the string. These are
+# Do not localize "CmdOrCtrl+;", or change the format of the string. These are
 # key identifiers, not messages displayed to the user.
 gotoLineModal.placeholder=Go to line…
-gotoLineModal.key=CmdOrCtrl+Shift+;
+gotoLineModal.key=CmdOrCtrl+;
```
Comment (flodolo) [inline assets/panel/debugger.properties]: This should have been caught in review (apparently I missed there were new strings in here)

Comment (jasonLaster) [reply assets/panel/debugger.properties]: good catch. I'll change this to `gotoLineModal.key2`

**Pull request merged**